### ISSUE TITLE
App header layout broken on mobile - App title pushed, drawer too narrow.

### ISSUE THREAD

**Issue by vedtam** [PolymerElements/app-layout#481]
Hi,

Sorry for being annoying with my bug reports, I am unable to use the app-layout on mobile because the (1) app title gets pushed down (see attached GIF), bellow the drawer toggle. The (2) drawer gets kinda narrow too, this could be possibly changed via CSS, but it still suck if this is a default behavior.

![app-header-layout-issue2](https://cloud.githubusercontent.com/assets/4981592/26273955/a06fbaf6-3d45-11e7-8bb3-afbadc84fc36.gif)

Freshly started project, code taken from app-layout samples:

```
<link rel="import" href="../../bower_components/polymer/polymer-element.html">
<link rel="import" href="../../bower_components/app-layout/app-layout.html">
<link rel="import" href="../../bower_components/paper-icon-button/paper-icon-button.html">
<link rel="import" href="../../bower_components/iron-icon/iron-icon.html">
<link rel="import" href="../../bower_components/iron-icons/iron-icons.html">


<dom-module id="webexpert-app">
  <template>
    <style>
      :host {
        display: block;
      }
      app-drawer-layout:not([narrow]) [drawer-toggle] {
        display: none;
      }
    </style>

    <!-- <iron-media-query query="max-width: 400px" query-matches="{{smallScreen}}"></iron-media-query> -->

    <app-drawer-layout force-narrow>
    <app-drawer slot="drawer">
      drawer-content
    </app-drawer>
    <app-header-layout>
      <app-header slot="header">
        <app-toolbar>
          <paper-icon-button icon="menu" drawer-toggle></paper-icon-button>
          <div main-title>App name</div>
        </app-toolbar>
      </app-header>

      main content

    </app-header-layout>
  </app-drawer-layout>

  </template>

  <script>
    /**
     * @customElement
     * @polymer
     */
    class WebexpertApp extends Polymer.Element {
      static get is() { return 'webexpert-app'; }
      static get properties() {
        return {
          prop1: {
            type: String,
            value: 'webexpert-app'
          }
        };
      }

      ready() {
        super.ready();
        // ...
      }
    }

    window.customElements.define(WebexpertApp.is, WebexpertApp);
  </script>
</dom-module>

```

Tested on Chrome 58, using Dev Tools.

**Comment by mathieuxxxx** [comment]
I'm also seeing the problem with the stock polymer-2-application from the polymer init.

The title gets pushed down, the toolbar disappear and the menu items always shows up.

**Comment by kdjlee** [comment]
Works for me.

<img width="556" alt="screen shot 2017-05-20 at 7 40 31 am" src="https://cloud.githubusercontent.com/assets/28826296/26276626/a1e54e3e-3d2f-11e7-8341-c042a726e221.png">

<img width="528" alt="screen shot 2017-05-20 at 7 40 21 am" src="https://cloud.githubusercontent.com/assets/28826296/26276627/a456353e-3d2f-11e7-84aa-f4101bece9f0.png">

Btw I see issue only when set Chrome emulation mobile to "iPhone 6" mode but it works fine on the real iPhone 6 and desktop Safari. Sounds to me a `polymer serve` issue.

**Comment by vedtam** [comment]
Indeed, it works on responsive, or the nexus viewport. Hope this will be only chrome/serve related. :)

**Comment by mathieuxxxx** [comment]
I'm serving my app with node.js/express and still have the issue in "Mobile" mode in Chrome Dev

![image](https://cloud.githubusercontent.com/assets/2973549/26276801/9dc846ae-3d4c-11e7-976e-84f3f5c39193.png)

**Comment by vedtam** [comment]
I was hoping that it won't become more interesting :D

**Comment by frankiefu** [comment]
Seems like styling is messed up (e.g. `@apply` is not setup correctly) in Chrome dev emulation "iPhone" mode. Maybe something to do with user agent detection code in ShadyCSS?  I see in console `ShadyCSS.nativeCss` returns `false`.

cc @sorvell @azakus

**Comment by azakus** [comment]
Chrome's iphone emulation changes the useragent to Safari 9, which has known issues with native CSS Custom Properties, causing us to disable them in ShadyCSS: https://github.com/webcomponents/shadycss/blob/a815c74a03e11b7458561fce50cc97766f0fd53b/src/style-settings.js#L24-L26
However, this interacts poorly with ShadyDOM, as shim'd CSS Custom Properties + native ShadowDOM is not supported.

The right fix would be to override this check if native ShadowDOM is used.

**Comment by azakus** [comment]
Published fix as ShadyCSS v1.0.1 and webcomponentsjs v1.0.1
